subiquity/tests/api/test_api.py

```diff
@@ -1651,3 +1651,59 @@ async def test_ad(self):
             }
             result = await instance.post(endpoint, ad_dict)
             self.assertEqual(['OK'], result)
+
+            ad_dict = {
```

Comment:
it would be nice to move some of this testing to unit tests instead of the API ones, but having at least one API test is still nice so the tests here won't actually be slower.  *shrug*  

The validation methods aren't part of the controller, so just calling those methods direct from unit test shouldn't be too bad.  An example of setting up a controller for unit test can be seen at subiquity/server/controllers/tests/test_filesystem.py  if you need it.

Reply:
Indeed, we can take advantage of those out-of-class routines to test them exaustively in unit tests and test the wiring-up in the API with just a few cases. I like the idea.